### PR TITLE
golang filter: don't track useless go.sum

### PR DESCRIPTION
**Pull request by spacewander**
`go.sum` is useless according to:
https://github.com/envoyproxy/envoy/pull/28595#discussion_r1276272831 

Add it to the gitignore so it won't be committed
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

Commit Message: don't track useless go.sum
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: contrib/golang/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+go.sum
```
Comment (phlax) [inline contrib/golang/.gitignore:1]: this doesnt seem correct - there are other `go.sum` files beneath this directory

also - i think if we are going to ignore something it should be in the main `.gitignore` at the root of the rpo

Comment (phlax) [reply contrib/golang/.gitignore:1]: correction - there are only `go.mod` files beneath this directory - but i think the point about not adding another .gitignore stands

Comment (spacewander) [reply contrib/golang/.gitignore:1]: @phlax 
What about using this?
```
/contrib/golang/**/test_data/go.sum
/contrib/golang/**/test_data/*/go.sum
```

Comment (phlax) [reply contrib/golang/.gitignore:1]: sgtm

Comment (phlax) [reply contrib/golang/.gitignore:1]: /wait

Comment (spacewander) [reply contrib/golang/.gitignore]: @phlax 
Solved in new commit

**Review by wangfakang — APPROVED** [review]
LGTM

**Review by phlax — APPROVED** [review]
lgtm, thanks @spacewander

**Pull request merged**